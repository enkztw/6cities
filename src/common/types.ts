export type property = `apartment` | `room` | `house` | `hotel`;
export type id = number
export type city = `Paris` | `Cologne` | `Brussels` | `Amsterdam` | `Hamburg` | `Dusseldorf`
export type sort = `rating` | `id` | `toLow` | `toHigh`


