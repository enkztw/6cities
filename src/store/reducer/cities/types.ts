enum Types {
  SET_CITY = `SET_CITY`
}

export default Types;
