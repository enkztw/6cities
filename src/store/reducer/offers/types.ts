enum Types {
  SET_OFFER = `SET_OFFER`,
  SET_COMMENTS = `SET_COMMENTS`,
  SET_OFFERS = `SET_OFFERS`,
  SET_FAVORITES = `SET_FAVORITES`,
  SET_NEARBY = `SET_NEARBY`,
  SET_SORTING = `SET_SORTING`
}

export default Types;
