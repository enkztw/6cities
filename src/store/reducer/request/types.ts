enum Types {
  SET_REQUEST = `SET_REQUEST`,
}

export default Types;

