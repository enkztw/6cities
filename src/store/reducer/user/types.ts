enum Types {
  SET_AUTH = `SET_AUTH`
}

export default Types;

