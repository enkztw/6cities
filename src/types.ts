export type property = `apartment` | `room` | `house` | `hotel`;

export type location = {
    latitude: number;
    longitude: number;
    zoom: number;
}

export type id = number
export type offers = Offer[]
export type cityName = `Paris` | `Cologne` | `Brussels` | `Amsterdam` | `Hamburg` | `Dusseldorf`

export interface City {
    location: location;
    name: cityName;
}

export interface User {
    readonly id: id;
    avatarUrl: string;
    isPro: boolean;
    name: string;
    email?: string;
    description?: string;
}

export interface Status {
    isAuthorized: boolean;
}

export interface Review {
    readonly id: id;
    comment: string;
    date: string;
    rating: number;
    user: User;
}

export interface Offer {
    readonly id: id;
    bedrooms: number;
    city: City;
    description: string;
    goods: string[];
    host: User;
    images: string[];
    isFavorite: boolean;
    isPremium: boolean;
    location: location;
    maxAdults: number;
    previewImage: string;
    price: number;
    rating: number;
    title: string;
    type: property;
}

