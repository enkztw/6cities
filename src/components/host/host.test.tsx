import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {User} from '../../types';
import Host from './host';


const host: User = {
  id: 3,
  isPro: true,
  name: `Angelina`,
  avatarUrl: `img/avatar-angelina.jpg`,
  description: `Hello!`
};

it(`<Host /> should render`, () => {
  const tree = renderer.create(<Host {...host} />).toJSON();

  expect(tree).toMatchSnapshot();
});
