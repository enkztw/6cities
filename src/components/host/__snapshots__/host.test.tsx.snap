// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Host /> should render 1`] = `
<div
  className="property__host"
>
  <h2
    className="property__host-title"
  >
    Meet the host
  </h2>
  <div
    className="property__host-user user"
  >
    <div
      className="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
    >
      <img
        alt="Host avatar"
        className="property__avatar user__avatar"
        height="74"
        src="img/avatar-angelina.jpg"
        width="74"
      />
    </div>
    <span
      className="property__user-name"
    >
      Angelina
    </span>
  </div>
  <div
    className="property__description"
  >
    <p>
      Hello!
    </p>
  </div>
</div>
`;
